### ISSUE TITLE
Some questions about interpret the code

### ISSUE THREAD

**Issue by Payuing** [AmirHoseinSafari/LRCN-drug-resistance#1]
These questions may sounds silly to you. I do want to confirm and get some heads up from you.
These may also help late followers, so I will sum these up once I figure them out.
Thanks.

- The source of bccdc data
-  What is feature importance?
- What does common elements do?
- How to generate the csv files mentioned in load_data.py? e.g. shuffled_operon_global.csv in load_data_operon_dataset
- What does pipe_line_gene do?
- What is ae_best_weights?
- What does wide_n_deep does?
- In model_one_vs_all, line 383 and line 406, why doing twice?
- model_CNN_LSTM_limited_1, is this model used in paper?
- What does get_model_GBT do?
- What does BO_GBT() do?

**Comment by AmirHoseinSafari** [comment]
Hi,

These are the answers to your questions, respectively:

1. This is mentioned in the final version of the paper that the BCCDC data is from another publication by our group "Geographic heterogeneity impacts drug resistance predictions in Mycobacterium tuberculosis."
2. You don't need to use "feature importance" since we didn't use it in the LRCn paper and it's for future works. My suggestion is to use the code prior to this commit on the git: 5db94335aec7fcdae81511994ff7bac350981703
3. Refer to the answer to the previous question.
4. Part of them are in the "dataset_creator" package and part of them are in the "Genotype-collector-and-SNP-dataset-creator" Repository.
5. "pipe_line_gene" is the main function for gene-based data that we used. You need to call any method that we have through this function.
6. Ignore it (refer to the answer to the second question).
7. This is mentioned in the final version of the paper. It is one of the state-of-the-art models that we used.
8. We didn't use it in the paper. It was just an experiment.
9. We didn't use it in the paper. It was just an experiment.
10. This is part of the Bayesian optimization process for the gradient boosting model. This function will get the hyperparameters as input and then build and train the GBT model.
11. This is part of the Bayesian optimization process for the gradient boosting model. This function decides the parameters that we are going to pass to the get_model_GBT function.
